source/geometry_model/ellipsoidal_chunk.cc

```diff
@@ -40,6 +42,326 @@ namespace aspect
 
+    template <int dim>
+    bool
+    EllipsoidalChunk<dim>::In2dPolygon(dealii::Point<2> &point,const std::vector<std::vector<double> > &pointList)
+    {
+      /**
+       * This code has been based on http://geomalgorithms.com/a03-_inclusion.html,
+       * and therefore requires the following copyright notice:
+       *
+       * Copyright 2000 softSurfer, 2012 Dan Sunday
+       * This code may be freely used and modified for any purpose
+       * providing that this copyright notice is included with it.
+       * SoftSurfer makes no warranty for this code, and cannot be held
+       * liable for any real or imagined damage resulting from its use.
+       * Users of this code must verify correctness for their application.
+       */
+      int pointNo = pointList.size();
+      int    wn = 0;    // the  winding number counter
+      int   j=pointNo-1;
+
+
+      // loop through all edges of the polygon
+      for (int i=0; i<pointNo; i++)
+        {
+          // edge from V[i] to  V[i+1]
+          if (pointList[j][1] <= point[1])
+            {
+              // start y <= P.y
+              if (pointList[i][1]  > point[1])      // an upward crossing
+                if (( (pointList[i][0] - pointList[j][0]) * (point[1] - pointList[j][1])
+                      - (point[0] -  pointList[j][0]) * (pointList[i][1] - pointList[j][1]) ) > 0)
+                  {
+                    // P left of  edge
+                    ++wn;            // have  a valid up intersect
+                  }
+            }
+          else
+            {
+              // start y > P.y (no test needed)
+              if (pointList[i][1]  <= point[1])     // a downward crossing
+                if (( (pointList[i][0] - pointList[j][0]) * (point[1] - pointList[j][1])
+                      - (point[0] -  pointList[j][0]) * (pointList[i][1] - pointList[j][1]) ) < 0)
+                  {
+                    // P right of  edge
+                    --wn;            // have  a valid down intersect
+                  }
+            }
+          j=i;
+        }
+      return (wn != 0);
+    }
+
+    /**
+     * the EllipsoidalChunkTopography class
+     */
+    // constructor
+    template <int dim>
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::EllipsoidalChunkTopography()
+    {}
+
+
+    template <int dim>
+    double
+    EllipsoidalChunk<dim>::EllipsoidalChunkTopography::value (const double lon,
+                                                              const double lat) const
+    {
+      double value = 0;
+
+      switch (topo_type)
+        {
+          case NO_TOPOGRAPHY:
+            return 0;
+            break;
+
+          case PRM_EXACT:
+            for (unsigned int i = 0; i < point_lists.size(); i++)
+              {
+                Point<2> p1(lon * 180/numbers::PI,lat * 180/numbers::PI);
+
+                if (In2dPolygon(p1, point_lists[i]))
+                  {
+                    value = topography_values[i];

```

Comment:
in this code, you set `value`, but you continue the loop. Do you expect that you may encounter another polygon in which the point may be? Or could you just return the value right away?